stockApp/stockApp/Protocols/AssetViewModelProtocol.swift

```diff
@@ -0,0 +1,9 @@
+import Foundation
+
+protocol AssetViewModelProtocol: AnyObject {
+    
+    var useCase: GetAssetUseCase { get set }
+    var observerData: (([Asset]) -> ())? { get set }
+    
+    func retrieveData(filter: String, value: String, onCompletion: @escaping () -> Void)
```

Comment:
```suggestion
    func retrieveData(filter: String, value: String)
```
give a proper name to the function. like `getAssets(for filter: String, value: String)` 

you variable and functions and class etc must be self-descritive